crates/rust-analyzer/src/config.rs

```diff
@@ -934,6 +939,16 @@ impl Config {
         }
     }
 
+    pub fn extra_env(&self) -> &FxHashMap<String, String> {
+        &self.data.cargo_extraEnv
+    }
+
+    pub fn check_on_save_extra_env(&self) -> FxHashMap<String, String> {
+        let mut extra_env = self.data.cargo_extraEnv.clone();
+        extra_env.extend(self.data.checkOnSave_extraEnv.clone());
+        extra_env
+    }
```

Comment:
I think we should fall back to cargo_extraEnv, not extend them (this is in line with the checkOnSave configs).

Reply:
Hm, actually I am not sure what would make more sense here

Reply:
I prefer this approach: using `extend` means that individual variables are overwritten, which matches how env vars work in general (i.e., you get the ambient env vars, then the `server.extraEnv` overwrites some of those, then `cargo.extraEnv` overwrites some of those and `checkOnSave.extraEnv` overwrites some of those).

If `checkOnSave.extraEnv` replaces `cargo.extraEnv`, then I suspect that folks will have to duplicate a lot of items to keep the two aligned (e.g., `PATH` and `CARGO_HOME` are likely to be the same, although I could see someone wanting to have a slightly different `RUSTFLAGS` to add more args for `checkOnSave.extraEnv`).

Reply:
Ye, I in this case extending does make more sense I agree, I have left it as was done in this PR.